packages/vizbuilder/src/helpers/chartconfig.js

```diff
@@ -0,0 +1,330 @@
+      config.body = d =>
+        `<div>
+  <p>${keyBody}: ${formatAbbreviate(d[keyBody])}</p>
+  <p>Year: ${d.Year}</p>
+  </div>`;
+    }
+  }
+  else {
+    if (query.moe) {
+      const moeName = query.moe.name;
+      config.body = d =>
+        `<div>
+<p>${keyBody}: ${formatAbbreviate(d[keyBody])}</p>
+<p>MOE: ±${formatAbbreviate(d[moeName])}</p>
+</div>`;
+    }
+    else {
+      config.body = d =>
+        `<div>
+<p>${keyBody}: ${formatAbbreviate(d[keyBody])}</p>
+</div>`;
+    }
+  }
+
+  return config;
+};
+
+export const timelineConfig = {
+  // time: !isOpen ? "ID Year" : ""
+};
+
+const makeConfig = {
+  barchart(commonConfig, query, flags) {
+    // BAR CHART
+    // Useful for discrete categories of data, specially integers.
+
+    const {timeDrilldown, drilldown, measure} = query;
+
+    const drilldownName = drilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      discrete: "x",
+      x: drilldownName,
+      xConfig: {title: drilldownName},
+      y: measureName,
+      yConfig: {title: measureName},
+      stacked: drilldown.depth > 1
+    };
+
+    if (timeDrilldown) {
+      config.groupBy = [timeDrilldown.hierarchy.levels[1].name];
+    }
+
+    if (flags.availableKeys.has("State")) {
+      config.xSort = sortByGeographyState;
+    }
+    else {
+      config.xSort = sortByCustomKey(drilldownName);
+    }
+
+    return config;
+  },
+  barchartyear(commonConfig, query) {
+    const {drilldown, timeDrilldown, measure} = query;
+
+    const drilldownName = timeDrilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      discrete: "x",
+      x: drilldownName,
+      xConfig: {title: drilldownName},
+      y: measureName,
+      yConfig: {title: measureName},
+      stacked: true,
+      groupBy: [drilldown.name],
+      tooltipConfig: tooltipConfig(query, drilldownName)
+    };
+
+    return config;
+  },
+  histogram(commonConfig, query, flags) {
+    // HISTOGRAM
+    // Allows to see the frequency distribution of a continuous dataset.
+    // Useful mainly with range buckets.
+
+    const config = this.barchart(commonConfig, query, flags);
+    config.groupPadding = 0;
+    return config;
+  },
+  donut(commonConfig, query) {
+    const {drilldown, measure} = query;
+
+    const drilldownName = drilldown.name;
+    const measureName = measure.name;
+
+    const config = {
+      ...commonConfig,
+      xConfig: {title: null},
+      y: measureName,
+      yConfig: {title: measureName},
+      groupBy: drilldownName
+    };
+
+    return config;
+  },
+  geomap(commonConfig, query, flags) {
+    const drilldownName = query.drilldown.name;
+
+    const config = {
+      ...commonConfig,
+      ...flags.colorScale,
+      tiles: false,
+      id: `ID ${drilldownName}`,
+      groupBy: `ID ${drilldownName}`,
+      ocean: "transparent",
+      projection: "geoAlbersUsa",
+      topojson: "/topojson/states.json",
```

Comment:
we need to generalize this. how about this solution:

The user of the `Vizbuilder` component can pass a lookup object to props that is a mapping of dimension names to topojson files. We will only draw Geomaps if we have a correct topojson mapping for the dimension selected. For example:

```jsx
<Vizbuilder src="https://canon-api.datausa.io"
  topojson={{
    State: "/topojson/states.json",
    County: "/topojson/counties.json"
  }} />
```

Reply:
Considering the need for chart config from the user, for now I set it as another attribute, `config`, which receives an object with 2 keys:
* `topojson`, whose purpose is provide this config
* `chartConfig`, which provides the config for the charts
The detail of the structure are available in the [`propTypes` property](https://github.com/Datawheel/datawheel-canon/blob/096b8f5a0900b84d4341cca21e35db479343399a/packages/vizbuilder/src/index.js#L121), and an example is available in the [`defaultProps` property](https://github.com/Datawheel/datawheel-canon/blob/096b8f5a0900b84d4341cca21e35db479343399a/packages/vizbuilder/src/index.js#L154), both in the main Vizbuilder component.